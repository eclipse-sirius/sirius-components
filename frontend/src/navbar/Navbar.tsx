/*******************************************************************************
 * Copyright (c) 2019, 2020 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import { useAuth } from 'auth/useAuth';
import { LoggedOutNavbar } from 'navbar/LoggedOutNavbar';
import { NavigationBar } from 'navbar/NavigationBar';
import React from 'react';

export const Navbar = () => {
  const { username } = useAuth() as any;
  if (username) {
    return <NavigationBar />;
  }
  return <LoggedOutNavbar />;
};
