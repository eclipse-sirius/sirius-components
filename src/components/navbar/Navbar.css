/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

.navbar {
  display: flex;
  flex-direction: row;

  padding: 0 var(--layoutDimension-m);

  height: var(--layoutDimension-xl);

  background-color: var(--white);
  box-shadow: var(--boxShadow-float);
}

.navbar-title {
  margin: auto 0;
}
