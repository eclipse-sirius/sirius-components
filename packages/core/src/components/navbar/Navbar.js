/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';

import { classNames } from '../../common/classnames';

import { Spacing } from '../spacing/Spacing';
import { S, M } from '../spacing/SpacingConstants';
import { Text } from '../text/Text';
import { BOLD, EXTRA_EXTRA_LARGE } from '../text/TextConstants';

import './Navbar.css';

const NAVBAR__CLASS_NAMES = 'navbar';
const NAVBAR_CONTAINER__CLASS_NAMES = 'navbar-container';
const NARBAR_TITLE__CLASS_NAMES = 'navbar-title title-xxl';

/**
 * The Navbar is used to display a navigation bar on top of the user interface.
 * It can contain the title of the application and various actions.
 */
export const Navbar = ({ className, ...props }) => {
  const navClassNames = classNames(NAVBAR__CLASS_NAMES, className);
  return (
    <nav className={navClassNames} {...props}>
      <Spacing top={S} right={M} bottom={S} left={M}>
        <div className={NAVBAR_CONTAINER__CLASS_NAMES}>
          <div className={NARBAR_TITLE__CLASS_NAMES}>
            <Text weight={BOLD} size={EXTRA_EXTRA_LARGE}>
              Eclipse Sirius
            </Text>
          </div>
        </div>
      </Spacing>
    </nav>
  );
};
