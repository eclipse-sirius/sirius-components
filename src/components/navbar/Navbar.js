/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import PropTypes from 'prop-types';

import { classNames } from '../../common/classnames';

import { IconMenu } from '../icons/IconMenu';
import { Spacing } from '../spacing/Spacing';
import { S, M } from '../spacing/SpacingConstants';
import { Text } from '../text/Text';
import { BOLD, EXTRA_EXTRA_LARGE } from '../text/TextConstants';

import './Navbar.css';

const NAVBAR__CLASS_NAMES = 'navbar';
const NAVBAR_CONTAINER__CLASS_NAMES = 'navbar-container';
const NARBAR_TITLE__CLASS_NAMES = 'navbar-title title-xxl';

const propTypes = {
  onHamburgerMenuClick: PropTypes.func.isRequired
};
const defaultProps = {
  onHamburgerMenuClick: () => {}
};

/**
 * The Navbar is used to display a navigation bar on top of the user interface.
 * It can contain the title of the application and various actions.
 */
export const Navbar = ({ className, onHamburgerMenuClick, ...props }) => {
  const navClassNames = classNames(NAVBAR__CLASS_NAMES, className);
  return (
    <nav className={navClassNames} {...props}>
      <Spacing top={S} right={M} bottom={S} left={M}>
        <div className={NAVBAR_CONTAINER__CLASS_NAMES}>
          <Spacing right={M}>
            <IconMenu onClick={onHamburgerMenuClick} />
          </Spacing>
          <div className={NARBAR_TITLE__CLASS_NAMES}>
            <Text weight={BOLD} size={EXTRA_EXTRA_LARGE}>
              Eclipse Sirius
            </Text>
          </div>
        </div>
      </Spacing>
    </nav>
  );
};
Navbar.propTypes = propTypes;
Navbar.defaultProps = defaultProps;
