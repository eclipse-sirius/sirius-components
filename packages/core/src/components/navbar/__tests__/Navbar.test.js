/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Renderer from 'react-test-renderer';

import { Navbar } from '../Navbar';

describe('Navbar', () => {
  it('renders the navbar component', () => {
    const navbar = Renderer.create(
      <MemoryRouter initialEntries={['/']}>
        <Navbar />
      </MemoryRouter>
    );
    expect(navbar.toJSON()).toMatchSnapshot();
  });
});
