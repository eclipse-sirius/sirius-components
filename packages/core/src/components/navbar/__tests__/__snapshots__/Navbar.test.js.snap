// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the navbar component 1`] = `
<nav
  className="navbar"
>
  <div
    style={
      Object {
        "paddingBottom": "8px",
        "paddingLeft": "16px",
        "paddingRight": "16px",
        "paddingTop": "8px",
      }
    }
  >
    <div
      className="navbar-container"
    >
      <div
        className="navbar-title title-xxl"
      >
        <div
          className="text xxl bold"
        >
          Eclipse Sirius
        </div>
      </div>
    </div>
  </div>
</nav>
`;
